pkg/filesystem/path/parser.go

```diff
@@ -0,0 +1,150 @@
+package path
+
+import (
+	"strings"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// Parser is used by Resolve to parse paths in the resolution.
+type Parser interface {
+	ParseScope(scopeWalker ScopeWalker) (next ComponentWalker, remainder RelativeParser, err error)
+}
+
+// RelativeParser is used by Resolve to parse relative paths in the resolution.
+type RelativeParser interface {
+	ParseFirstComponent(componentWalker ComponentWalker, mustBeDirectory bool) (next GotDirectoryOrSymlink, remainder RelativeParser, err error)
+}
+
+type unixAbsoluteParser struct {
```

Comment:
Maybe just call this unixParser, for consistency with Parser?

Do we want to place all of this code in a `unix_parser.go`, so that the Windows and UNIX implementations aren't all smacked in the same source file?

Reply:
Check!